presto-main/src/main/java/io/prestosql/server/security/UserMapping.java

```diff
@@ -84,33 +86,57 @@ public UserMappingRules(
         }
     }
 
+    enum Case
+    {
+        KEEP(Function.identity()),
+        LOWER(value -> value.toLowerCase(Locale.ENGLISH)),
+        UPPER(value -> value.toUpperCase(Locale.ENGLISH)),
+        /**/;
+
+        private final Function<String, String> transformation;
+
+        Case(Function<String, String> transformation)
+        {
+            this.transformation = requireNonNull(transformation, "transformation is null");
+        }
+
+        public final String transform(String value)
+        {
+            return transformation.apply(value);
+        }
+    }
+
     public static final class Rule
     {
         private final Pattern pattern;
         private final String user;
         private final boolean allow;
+        private final Case kase;
 
         public Rule(String pattern)
         {
-            this(pattern, "$1", true);
+            this(pattern, "$1", true, Case.KEEP);
         }
 
         @JsonCreator
         public Rule(
                 @JsonProperty("pattern") String pattern,
                 @JsonProperty("user") Optional<String> user,
-                @JsonProperty("allow") Optional<Boolean> allow)
+                @JsonProperty("allow") Optional<Boolean> allow,
+                @JsonProperty("case") Optional<Case> kase)
         {
             this(pattern,
                     requireNonNull(user, "user is null").orElse("$1"),
-                    requireNonNull(allow, "allow is null").orElse(TRUE));
+                    requireNonNull(allow, "allow is null").orElse(TRUE),
+                    requireNonNull(kase, "kase is null").orElse(Case.KEEP));
```

Comment:
Case? Like class.

Reply:
parameter name is `kase` because `case` is a keyword. The thing commonly used in other projects is something like `_case` or `case_` but here checkstyle disallows it, so changing spelling is second commonly used thing (like `class` -> `klass` or `clazz`)

Reply:
uh... sorry auto correction fix `caze` to `case`. I wanted to say:

`caze`? like `clazz`? `kase` looks like a typo to me. Also I am under impression that it is more common in languages that `z` sounds closer to `s` than `k` to `c`.

I don't have strong preference. Up to you.

Reply:
I am not sure about this regexp. Please open such pull request and we could continue the discussion there. How about applying: https://github.com/prestosql/presto/pull/4736/files#r467496420?